3506/0/ruby_koans/triangle.rb

```diff
@@ -0,0 +1,29 @@
+# Triangle Project Code.
+
+# Triangle analyzes the lengths of the sides of a triangle
+# (represented by a, b and c) and returns the type of triangle.
+#
+# It returns:
+#   :equilateral  if all sides are equal
+#   :isosceles    if exactly 2 sides are equal
+#   :scalene      if no sides are equal
+#
+# The tests for this method can be found in
+#   about_triangle_project.rb
+# and
+#   about_triangle_project_2.rb
+#
+def triangle(a, b, c)
+  # WRITE THIS CODE
+  a, b, c = [a, b, c].sort
+  raise TriangleError if a <= 0
+  raise TriangleError if a + b <= c
+
+  return :equilateral if a == b && b == c
```

Comment:
Why we need these conditions if we already have sorted array? Use Array methods for moving out from complex conditions